insight/pv_model/pv_model_class.py

```diff
@@ -0,0 +1,260 @@
+import numpy as np
+import scipy as sp
+from scipy import constants
+import matplotlib.pyplot as plt
+
+
+class PvModel:
+    def __init__(self, Iscn, Vocn, Imp, Vmp, Kv, Ki, Ns, Gn, G, Tn, T, Egap, err, inverter, array_dim):
+        # Initializes all standard values
+        self.Iscn = Iscn
+        self.Vocn = Vocn
+        self.Imp = Imp
+        self.Vmp = Vmp
+        self.Kv = Kv
+        self.Ki = Ki
+        self.Ns = Ns
+        self.Gn = Gn
+        self.G = G
+        self.Tn = Tn
+        self.T = T
+        self.Egap = Egap
+        self.err = err
+        self.inverter = inverter
+        # array with [number of strings, number in series (per string)]
+        self.array_dim = array_dim
+
+        self.q = constants.e
+        self.k = constants.Boltzmann
+
+    def calculate_currents(self, Rs, Rp, a, Iscn, Vocn, Ki, Kv, Vt, Ns, G, Gn, deltaT):
+        """Calculates and returns the different current values of interest. Takes
+        different PV module specifications and weather data as input.
+        """
```

Comment:
A nice way to further document your code is to include documentation on the input and output parameters for different functions (it's also nice to do this for errors that can be raised, but we don't include that in our code yet). I believe one of the standards for doing this is through `Sphinx`, which is a documentation generator used in Python. Rather than diving into all of `Sphinx`'s documentation, check out the way we handled this documentation at BE [here](https://github.com/Breakthrough-Energy/PreREISE/blob/develop/prereise/gather/demanddata/nrel_efs/aggregate_demand.py). 

For example, the doc string for `Vocn` would look something like this:

`:param float Vocn: The rated open-circuit voltage of the specified PV mocule.`